core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2793,7 +2771,7 @@ object Stream {
           val runStep =
             Algebra
               .compileScope(
-                scope,
+                scope.asInstanceOf[fs2.internal.CompileScope[F, UO]], // todo: resolve cast
```

Comment:
Can the TODO be resolved by specifying UO as type param to getScope (instead of Nothing)?

Reply:
Unfortunatelly the result of `unconsAsync` is `Pull[F, Nothing, AsyncPull[F, Option[(Segment[O, Unit], Stream[F, O])]]]` so I would need then typecase UO back to Nothing, at least that's what compiler is asking me to do. 
I have plan to address in #1056.